File: DSA/NabhanshRishiGaur_590028370/Day_8/Question_1.c
bool isPalindrome(char* s) {
    int l = 0;
    int r = strlen(s) - 1;

    while (l < r) {
        while (l < r &&
              !((s[l] >= '0' && s[l] <= '9') ||
                (s[l] >= 'A' && s[l] <= 'Z') ||
                (s[l] >= 'a' && s[l] <= 'z')))
            l++;

        while (l < r &&
              !((s[r] >= '0' && s[r] <= '9') ||
                (s[r] >= 'A' && s[r] <= 'Z') ||
                (s[r] >= 'a' && s[r] <= 'z')))
            r--;

        char c1 = s[l];
        char c2 = s[r];

        if (c1 >= 'A' && c1 <= 'Z') c1 += 32;
        if (c2 >= 'A' && c2 <= 'Z') c2 += 32;

        if (c1 != c2)
            return false;

        l++;
        r--;
    }

    return true;
}

File: DSA/ShlokMadan_590017851/Day_8/Question1.java
public class Question1 
{
    public static boolean isPalindrome(String s) 
    {
        int left = 0;
        int right = s.length() - 1;

        while (left < right) {

            while (left < right && !Character.isLetterOrDigit(s.charAt(left))) 
                {
                left++;
            }

            while (left < right && !Character.isLetterOrDigit(s.charAt(right))) 
                {
                right--;
            }

            if (Character.toLowerCase(s.charAt(left)) !=
                Character.toLowerCase(s.charAt(right))) 
                {
                return false;
            }

            left++;
            right--;
        }

        return true;
    }
    public static void main(String[] args) 
    {
        String s = "A man, a plan, a canal: Panama";
        System.out.println(isPalindrome(s));
    }
}